pkg/controller/import-controller.go

```diff
@@ -771,31 +843,160 @@ func createImportPodNameFromPvc(pvc *corev1.PersistentVolumeClaim) string {
+func makeNodeImporterPodSpec(args *importerPodArgs) *corev1.Pod {
+	// importer pod name contains the pvc name
+	podName, _ := args.pvc.Annotations[AnnImportPod]
+
+	volumes := []corev1.Volume{
+		{
+			Name: "shared-volume",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: DataVolName,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: args.pvc.Name,
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+
+	importerContainer := makeImporterContainerSpec(args.image, args.verbose, args.pullPolicy)
+
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: args.pvc.Namespace,
+			Annotations: map[string]string{
+				AnnCreatedBy: "yes",
+			},
+			Labels: map[string]string{
+				common.CDILabelKey:       common.CDILabelValue,
+				common.CDIComponentLabel: common.ImporterPodName,
+				common.PrometheusLabel:   "",
+			},
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:            "init",
+					Image:           args.image,
+					ImagePullPolicy: corev1.PullPolicy(args.pullPolicy),
+					Command:         []string{"sh", "-c", "cp /usr/bin/cdi-containerimage-server /shared/server"},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/shared",
+							Name:      "shared-volume",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				*importerContainer,
+				{
+					Name:            "server",
+					Image:           args.importImage,
+					ImagePullPolicy: corev1.PullPolicy(args.pullPolicy),
+					Command:         []string{"/shared/server", "-p", "8100", "-image-dir", "/disk", "-ready-file", "/shared/ready", "-done-file", "/shared/done"},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/shared",
+							Name:      "shared-volume",
+						},
+					},
+				},
+			},
+			RestartPolicy:     corev1.RestartPolicyOnFailure,
+			Volumes:           volumes,
+			NodeSelector:      args.workloadNodePlacement.NodeSelector,
+			Tolerations:       args.workloadNodePlacement.Tolerations,
+			Affinity:          args.workloadNodePlacement.Affinity,
+			PriorityClassName: args.priorityClassName,
+		},
+	}
+
+	/**
+	FIXME: When registry source is ImageStream, if we set importer pod OwnerReference (to its pvc, like all other cases),
+	for some reason (OCP issue?) we get the following error:
+		Failed to pull image "imagestream-name": rpc error: code = Unknown
+		desc = Error reading manifest latest in docker.io/library/imagestream-name: errors:
+		denied: requested access to the resource is denied
+		unauthorized: authentication required
+	When we don't set pod OwnerReferences, all works well.
```

Comment:
see where you delete imagestream finalizer but not add it

Reply:
We add it [here](https://github.com/kubevirt/containerized-data-importer/pull/1913/commits/b942c8d6bc9c7bad4cc27c41769838b55ee13cd7#diff-868af4b730094e9180f78af4cc069ae6cecf7077a46c186553f331dccbd32993R515-R523)